[docs] Revert FetchContent section

ras0219-msft opened the pull request:
Because we do not officially support or test FetchContent, we should clarify for users that it is likely to break at any time and should not be used in production.

ras0219-msft committed:
[docs] Mark FetchContent as experimental

BillyONeal commented:
As written I don't understand this, we are documenting something and telling people not to do it? Why document it in the first place then? Being marked an experimental feature would only be appropriate if we were expecting that to become an official distribution channel, and we aren't going to do that.

I think we should just say "oops, we shouldn't have merged https://github.com/microsoft/vcpkg/pull/27158 " back that out, and consider this scenario when we get our deployment mechanisms straight.

Neumann-A commented:
> Because we do not officially support or test FetchContent, we should clarify for users that it is likely to break at any time and should not be used in production.

How is this different from a git submodule?

autoantwort commented:
Meanwhile https://discord.com/channels/400588936151433218/687365466422902841/885614141153349682

BillyONeal commented:
I asked the docs v-team (@ras0219-msft and @AugP) to take a look at this.

ras0219-msft committed:
[docs] Revert 'FetchContent' section

It is not a supported, designed, or tested method and has several significant shortcomings.

anthonyalayo commented:
> I tried out to use vcpkg via `FetchContent` and IMO it is better to add vcpkg as a git submodule because every time you do a "clean re-configure" by removing the `build` folder, vcpkg has to be cloned again wich might take some time and maybe install things like CMake or Ninja. Furthermore, I ran into issues when I tried to use a different generator than `Ninja`.

CMake 3.24 has `--fresh`, which does the "clean re-configure" without removing downloaded dependencies. Can we re-visit adding this back to the docs?
